ControlSettings/ControlSettings.cpp

```diff
@@ -248,16 +248,29 @@ namespace Plugin {
                , m_isDisabledHdmiIn4KZoom (false)
     {
         LOGINFO("Entry\n");
-        devicePtr = new WPEFramework::Plugin::TV;
+#ifndef ENABLE_TV_SUPPORT
+        devicePtr = new STB;
+#else
+        devicePtr = new TV;
+#endif
+
 	ControlSettings::_instance = this;
 	InitializeIARM();
 
+        if(devicePtr->isTvSupportEnabled())//TV Specific API
```

Comment:
This we diiscuss to move to TV and STB specifc class. so you should add virtial Register method

Reply:
We cant move out API registration to derive class because,

1)AS creating object for ControlSettings class which is independent of Device/STB/TV class. As a result, AS has access to an API that is only available in the ControlSettingsPlugin class scope.

2)Let's say you want to forward the API registration to the TV class.

registerMethod("getBacklight", &TVMgr::getBacklight, this, {2});----->API will be register for TV class which not available in AS scope.

If you want to register any API with the TV/STB/Device class, the second argument must be a function pointer, and the third argument must be "this" ptr.

In this scenario, AS will not call getBacklight since it is registered in the TV class, and AS can only view the controlSettingsplugin functions.

As a result, just the registration and init sequences were pushed into the Main ControlSettings class, and the functionality was moved to a TV/STB class.